wgpu-core/src/resource.rs

```diff
@@ -785,6 +836,28 @@ crate::impl_parent_device!(Buffer);
 crate::impl_storage_item!(Buffer);
 crate::impl_trackable!(Buffer);
 
+impl WebGpuError for CreateBufferError {
+    fn webgpu_error_type(&self) -> ErrorType {
+        let e: &dyn WebGpuError = match self {
+            Self::Device(e) => e,
+            Self::AccessError(e) => e,
+            Self::MissingDownlevelFlags(e) => e,
+
+            Self::IndirectValidationBindGroup(_) => {
+                return ErrorType::DeviceLost {
+                    reason: DeviceLostReason::Unknown,
+                }
+            }
```

Comment:
I think we should propagate the inner `DeviceError` since HAL could return an OOM or device loss when we try to create the bind groups.